staff/groups/aleti/github-app-2.0.0-react/src/App.js

```diff
@@ -0,0 +1,70 @@
+import React, { Component } from 'react';
+import SearchUsers from './components/SearchUsers'
+import UsersList from './components/UsersList'
+import UserDetails from './components/UserDetails'
+import logic from './logic/index'
+
+import './App.css';
+
+class App extends Component {
+  state = {
+    users: [],
+    user: {},
+    query: '',
+  }
+
+  _searchUsers = e => {
+    e.preventDefault();
+    if (this.state.query.length > 0) {
+      logic.searchUsers(this.state.query)
+        .then(users => {
+          this.setState({
+            users: users,
+            user: {},
+            query: ''
+          })
+        });
+    }
+  }
+
+  _retrieveUser = username => {
+    logic.retrieveUser(username)
+      .then(user => {
+        this.setState({
+          user: user,
+          query: ''
+        })
+      });
+  }
+
+  handleChange = e => {
```

Comment:
TIP following comments before, move this query handling to SearchUsers

Reply:
+1